dbcon/joblist/tuple-bps.cpp

```diff
@@ -2678,7 +2672,12 @@ void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
         }
     }
 
-    // Bug 3136, let mini stats to be formatted if traceOn.
+    if (memSizeForOutputRG)
+    {
+        fRm->returnMemory(memSizeForOutputRG, fJoinMemLimit);
```

Comment:
You have at least 3 places where these two statements are called. It is worth to make it a method or a function.

Reply:
or add
```
if (!amount)
    return;

```
To returnMemory